forem/forem
# refactor html classes, remove css rules, implement modal for tag history

## What type of PR is this? (check all applicable)


## Description

Using a modal to display expanded tag adjustment history for tags with more than 2 adjustments. 

This also includes some fast follow styling tweaks to lean more on Crayons utility HTML as opposed to custom CSS rules.

Modal is currently working and styled:
<img width="1314" alt="Screenshot 2023-09-24 at 11 29 28 PM" src="https://github.com/forem/forem/assets/7056287/a3a6ec6f-65f5-48f7-a3bb-fbabf677cd3d">
"See all" button in mod actions panel will show all previous tag adjustments.

<img width="1335" alt="Screenshot 2023-09-24 at 11 29 41 PM" src="https://github.com/forem/forem/assets/7056287/1a324941-8d7d-4d54-bee3-a7d4274b9181">
Modal overlays entire screen with styling per [designs](https://www.figma.com/file/yhg9HTZDFIeWuvBt9ygrss/Moderation?type=design&node-id=1733-2636&mode=design)

## Related Tickets & Documents


- Closes #19823

## QA Instructions, Screenshots, Recordings

On any article page (also visitable via /mod or the mod page for a tag) click into the Adjust Tag menu in the mod actions panel (shield icon opens the panel) and make sure the article has more than two adjustments. You can do this by adding or removing tags - I've been using tags like `ruby`, `java`, `javascript`, and `discuss` as I make adjustments.

Once an article has enough adjustments, you will see a "See all" button appear to the right of the "Previous tag adjustments" heading text. Clicking the button opens a modal which shows the entire history of tag adjustments on that article in reverse chronological order.

### UI accessibility checklist
_If your PR includes UI changes, please utilize this checklist:_

_For more info, check out the
[Forem Accessibility Docs](https://developers.forem.com/frontend/accessibility)._

## Added/updated tests?
_We encourage you to keep the code coverage percentage at 80% and above._

      have not been included_

## [optional] Are there any post deployment tasks we need to perform?

## [optional] What gif best describes this PR or how it makes you feel?

![person watering plant which grows into a tall giraffe and walks away whimsically](https://media1.giphy.com/media/KZHA5vftxNpHnVFWNw/giphy.gif)
